File: src/android/input/StartInput.java
package com.bandyer.cordova.plugin.input;

import org.json.JSONArray;
import org.json.JSONObject;

import com.bandyer.cordova.plugin.Constants;

public class StartInput {

    private String mUserAlias;

    public static StartInput createFrom(JSONArray argsArray) throws PluginInputNotValidException {
        StartInput startInput = new StartInput();
        try {
            JSONObject args = (JSONObject) argsArray.get(0);
            String username = args.getString(Constants.ARG_USER_ALIAS);
            if(username == null || username.equals("")) {
                throw new PluginInputNotValidException(Constants.ARG_USER_ALIAS + " cannot be null");
            }
            startInput.setUserAlias(username);

            return startInput;
        }catch (Throwable t) {
            throw new PluginInputNotValidException("error on StartInput " + t.getMessage(), t);
        }
    }

    public String getUserAlias() {
        return mUserAlias;
    }

    public void setUserAlias(String userAlias) {
        mUserAlias = userAlias;
    }
}

File: src/android/cordova_plugin/src/main/java/com/bandyer/cordova/plugin/input/StartInput.kt
package com.bandyer.cordova.plugin.input

import org.json.JSONArray
import org.json.JSONObject

import com.bandyer.cordova.plugin.Constants

class StartInput {

    var userAlias: String? = null

    companion object {

        @Throws(PluginInputNotValidException::class)
        fun createFrom(argsArray: JSONArray): StartInput {
            val startInput = StartInput()
            try {
                val args = argsArray.get(0) as JSONObject
                val username = args.getString(Constants.ARG_USER_ALIAS)
                if (username == null || username == "") {
                    throw PluginInputNotValidException(Constants.ARG_USER_ALIAS + " cannot be null")
                }
                startInput.userAlias = username

                return startInput
            } catch (t: Throwable) {
                throw PluginInputNotValidException("error on StartInput " + t.message, t)
            }

        }
    }
}
